pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -46,3 +70,426 @@ func (s *discovery) Close() {}
 func (s *discovery) handleStatusEndpoint(w http.ResponseWriter, _ *http.Request) {
 	_, _ = w.Write([]byte("Discovery Module is running"))
 }
+
+func (s *discovery) handleError(w http.ResponseWriter, route string, status int, err error) {
+	_ = log.Errorf("failed to handle /discovery/%s (status: %d): %v", route, status, err)
+	w.WriteHeader(status)
+}
+
+func (s *discovery) handleOpenPorts(w http.ResponseWriter, _ *http.Request) {
+	listeners, err := getListeners()
+	if err != nil {
+		s.handleError(w, pathOpenPorts, http.StatusInternalServerError, fmt.Errorf("failed to get open ports: %v", err))
+		return
+	}
+
+	resp := &model.OpenPortsResponse{
+		Listeners: *listeners,
+	}
+	utils.WriteAsJSON(w, resp)
+}
+
+var allowedEnvironmentVariables = map[string]struct{}{
+	"PATH":                     {},
+	"PWD":                      {},
+	"GUNICORN_CMD_ARGS":        {},
+	"WSGI_APP":                 {},
+	"DD_SERVICE":               {},
+	"DD_TAGS":                  {},
+	"DD_INJECTION_ENABLED":     {},
+	"SPRING_APPLICATION_NAME":  {},
+	"SPRING_CONFIG_LOCATIONS":  {},
+	"SPRING_CONFIG_NAME":       {},
+	"SPRING_PROFILES_ACTIVE":   {},
+	"CORECLR_PROFILER_PATH":    {},
+	"CORECLR_ENABLE_PROFILING": {},
+	"JAVA_TOOL_OPTIONS":        {},
+	"_JAVA_OPTIONS":            {},
+	"JDK_JAVA_OPTIONS":         {},
+	"JAVA_OPTIONS":             {},
+	"CATALINA_OPTS":            {},
+	"JDPA_OPTS":                {},
+	"VIRTUAL_ENV":              {},
+}
+
+func filterEnv(in []string) (out []string) {
+	for _, env := range in {
+		split := strings.SplitN(env, "=", 2)
+		if len(split) != 2 {
+			continue
+		}
+
+		name := split[0]
+		if _, ok := allowedEnvironmentVariables[name]; ok {
+			out = append(out, env)
+		}
+	}
+	return
+}
+
+func (s *discovery) handleGetProc(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	pidStr := vars["pid"]
+	pid, err := strconv.ParseUint(pidStr, 10, 32)
+	if err != nil {
+		s.handleError(w, pathGetProc, http.StatusBadRequest, fmt.Errorf("failed to convert pid to integer: %v", err))
+		return
+	}
+
+	if _, err := os.Stat(path.Join(procfs.DefaultMountPoint, pidStr)); os.IsNotExist(err) {
```

Comment:
The `procfs` path that DD uses can be different from `/proc` and can be set by config. You should use `kernel.HostProc` for this: https://github.com/DataDog/datadog-agent/blob/main/pkg/util/kernel/fs.go#L65

Reply:
I've removed this code in refactoring. I think the other stuff should be fine since gopsutil respects `HOST_PROC` and the agent init sets it if necessary?